server/routes/basicDetails.ts

```diff
@@ -0,0 +1,411 @@
+    return addressList.find(address => address.addressId === addressIdentifierNumber)
+  }
+
+  async function renderWarningTypes(
+    breachNotice: BreachNotice,
+    res: Response,
+    warningTypeRadioButtons: Array<RadioButton>,
+    errorMessages: ErrorMessages,
+    sentenceTypeSelectItems: Array<SelectItem>,
+  ) {
+    res.render('pages/warning-type', {
+      errorMessages,
+      breachNotice,
+      warningTypeRadioButtons,
+      sentenceTypeSelectItems,
+    })
+  }
+
+  router.get('/basic-details', async (req, res, next) => {
+    await auditService.logPageView(Page.BASIC_DETAILS, { who: res.locals.user.username, correlationId: req.id })
+    res.render('pages/basic-details')
+  })
+
+  // if this page hasnt been saved we want to go through and apply defaults otherwise dont do this
+  function checkBreachNoticeAndApplyDefaults(breachNotice: BreachNotice, basicDetails: BasicDetails) {
+    // we havent saved the page yet so default
+    if (!breachNotice.basicDetailsSaved) {
+      // load offender name from integration details
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.titleAndFullName = `${basicDetails.title} ${basicDetails.name.forename} ${basicDetails.name.middleName} ${basicDetails.name.surname}`
+      // default the radio buttons to true
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.useDefaultAddress = true
+      // eslint-disable-next-line no-param-reassign
+      breachNotice.useDefaultReplyAddress = true
+    }
+  }
+
+  function initiateAlternateAddressSelectItemList(
+    basicDetails: BasicDetails,
+    breachNotice: BreachNotice,
+  ): SelectItem[] {
+    const alternateAddressSelectItemList: SelectItem[] = [
+      {
+        text: 'Please Select',
+        value: '-1',
+        selected: true,
+      },
+      ...basicDetails.addresses.map(address => ({
+        text: [
+          address.buildingName,
+          address.addressNumber.concat(` ${address.streetName}`).trim(),
+          address.district,
+          address.townCity,
+          address.county,
+          address.postcode,
+        ]
+          .filter(item => item)
+          .join(', '),
+        value: `${address.addressId}`,
+        selected: false,
+      })),
+    ]
+
+    if (breachNotice.basicDetailsSaved) {
+      alternateAddressSelectItemList.forEach((selectItem: SelectItem) => {
+        if (breachNotice.offenderAddress && breachNotice.offenderAddress.addressId) {
+          if (breachNotice.offenderAddress.addressId !== -1) {
+            const selectItemValueNumber: number = +selectItem.value
+            if (breachNotice.offenderAddress.addressId === selectItemValueNumber) {
+              // eslint-disable-next-line no-param-reassign
+              selectItem.selected = true
+            } else {
+              // eslint-disable-next-line no-param-reassign
+              selectItem.selected = false
+            }
+          }
+        }
+      })
+    }
```

Comment:
Do you need to loop through the list a second time?  Could you set the selected flag in the original map call?

This would avoid you having to disable linting as well, and keep the fields immutable.  Hopefully you can do the same for the other list function too.

I think you could also get rid of the `basicDetailsSaved` flag too.

E.g.
```suggestion
        selected: address.addressId == breachNotice.offenderAddress?.addressId,
      })),
    ]
```

Reply:
Details saved is used for something else too. It drives navigation and which sections are editable on the check your report page.

Reply:
Okay, makes sense.  I think it still makes sense to move the setting of the selected flag into the first map loop if that's possible though.